README.md

```diff
@@ -1,81 +1,71 @@
+- **CheckTests**: Checks for the presence of test directories or test files.
+- **CheckAlmanack**: Runs the [Software Gardening Almanack](https://github.com/software-gardening/almanack) analysis.
 
-2. **CheckReadme**: Checks the cloned repository for the presence of a README file. It looks for various common README file names and reports whether one was found.
+The final output is a **consolidated CSV report** where each row represents a tool (i.e., a repository) with the following columns:
 
-3. **CheckDependencies**: Scans the repository for dependency files associated with different programming languages. It reports the presence of files for Python, JavaScript/Node.js, Java, and R.
+```Tool, CloneRepository, CheckReadme, CheckDependencies, CheckTests, Almanack```
 
-4. **CheckTests**: Looks for test directories or files within the repository.
+Each column shows the status (`PASS`/`FAIL`) for the respective check.
 
-5. **CheckAlmanack**: Implements the [Software Gardening Almanack](https://github.com/software-gardening/almanack) to gather various metrics about the repository.
-   - **Note:** The `CheckAlmanack` process now uses `python3` to execute the Python command. If you encounter issues, ensure that Python 3 is installed and that `python3` is in your system's PATH.
+## Setup & Usage with Docker
 
-6. **SaveToSynapse** (Optional): Uploads workflow results of the toolkit to a specified Synapse folder if the upload option is enabled.
+All dependencies are bundled in the Docker images built via a multi-stage Dockerfile. This workflow is designed to be run using Docker.
 
-## Setup
+### Building the Docker Images
 
-Install Nextflow:
+This project uses a multi-stage Docker build to produce two images:
 
-```sh
-curl -s https://get.nextflow.io | bash
-```
-
-Install Almanack:
-
-```sh
-pip install almanack
-```
-
-## Configuration
-
-You can configure the workflow using `nextflow.config`. Set your working directory here.
-
-## Usage
+- **Nextflow Image**: Contains Nextflow and common dependencies.
+- **Almanack Image**: Contains the Almanack tool for the dedicated process.
 
-To run the workflow, you need to provide the URL of the Git repository you want to analyze as a parameter. Here's how you can execute the workflow:
+Build the images using the commands below:
 
 ```bash
-nextflow run main.nf --repo_url <repository-url>
-```
-
-Replace `<repository-url>` with the URL of the Git repository you wish to check.
+# Build the Nextflow image (for running the overall workflow)
+docker build --target nextflow -t cckp-toolkit .
 
-## Example
+# Build the Almanack image (for the RunAlmanack process)
+docker build --target almanack -t cckp-toolkit-almanack .
+```
+## Running the Workflow
+You can run the workflow using Docker. You can either analyze a single tool by specifying its repository URL or run multiple tools using a sample sheet (a CSV file with a header repo_url).
 
+### Run with a Single Repository URL
```

Comment:
In don't think there are many (any?) cases where you would want to run nextflow inside a container. Aside from the docker-in-docker complications the great advantage of Nextflow is that it takes care of the dependencies by using process specific containers or environments.

I would update this section starting at 'Running the workflow' to just say 'Install nextflow' (link to to nextflow.io instructions page) and then the nextflow command to run rather than using a entrypoint.

eg 

```
nextflow run mc2-center/cckp-toolkit-workflow --sample_sheet <samplesheet>
```